sstables/consumer.hh

```diff
@@ -138,8 +144,10 @@ private:
                 data.trim_front(len);
                 return read_status::ready;
             } else {
-                _read_bytes = make_tracked_temporary_buffer(temporary_buffer<char>(len), _permit);
-                std::copy(data.begin(), data.end(), _read_bytes.get_write());
+                _read_bytes = std::vector<temporary_buffer<char>>();
+                _read_bytes.push_back(make_tracked_temporary_buffer(temporary_buffer<char>(len), _permit));
+                std::copy(data.begin(), data.end(), _read_bytes.front().get_write());
```

Comment:
Note: `data` itself is a temporary_buffer, so instead of a copy you can use temporary_buffer::share().

It can make a nice follow-up.

Reply:
We need to make it tracked though, but I think it is possible to add a new deleter to the deleter chain.

Reply:
Actually, we shouldn't be using temporary_buffer for something that has a fixed maximum size. We should read into a nice array of char8_t.

Reply:
Ah, it doesn't really matter. This only kicks in if the vint spans two disk buffers, which is super rare. So it doesn't matter if we're inefficient here.